engine/apps/schedules/models/custom_on_call_shift.py

```diff
@@ -705,10 +704,6 @@ def create_or_update_last_shift(self, data):
         instance_data["start_rotation_from_user_index"] = start_rotation_from_user_index
 
         if self.last_updated_shift is None or self.last_updated_shift.event_is_started:
-            # create new shift
```

Comment:
There is no need to remove this comment as it is related to the code below.
Also I think, we should set the current's shift name here or exclude the "name" from the `fields_to_remove` above, if we want shifts belonging to the same rotation to have the same name